### PR TITLE
Allow only trusted images and repositories.

### PR DESCRIPTION
**Pull request by dekiel**
**Description**

### <samp>🤖 Generated by Copilot at e3562f0</samp>

This pull request adds new constraint templates and resources to enforce a list of allowed image URLs for Pods in different namespaces and clusters using OPA and Gatekeeper. The constraints use the `k8sallowedimages` template defined in `opa/gatekeeper/constraint-templates/k8sAllowedImages.yaml` and set the enforcement action to `warn`. The constraints are applied to the Prow application, supporting tools, and prowjobs in the `prow/cluster/resources/gatekeeper-constraints` directory.

Changes proposed in this pull request:

### <samp>🤖 Generated by Copilot at e3562f0</samp>

*  Add a new constraint template `k8sallowedimages` for enforcing allowed image URLs and namespaces for Pods using OPA and Gatekeeper ([link](https://github.com/kyma-project/test-infra/pull/7882/files?diff=unified&w=0#diff-219a468056913f41307647bb5c44f3bdfa39f6073b39330478f3eee59dbcb074R1-R52))
*  Apply the `k8sallowedimages` template to the `default` namespace in the Prow cluster and specify the image URLs for the Prow application components ([link](https://github.com/kyma-project/test-infra/pull/7882/files?diff=unified&w=0#diff-f17f06783ce7e68c1840feb242b483daa40bb2607932c97d7db0b2398d9dc3b3R1-R33))
*  Apply the `k8sallowedimages` template to four other namespaces in the Prow cluster and specify the image URLs for the supporting tools ([link](https://github.com/kyma-project/test-infra/pull/7882/files?diff=unified&w=0#diff-3b21df708b9303d6ceeb5259b9c2846fba0d6985930fab0a24c2f87f226c556dR1-R28))
*  Apply the `k8sallowedimages` template to the `default` namespace in the Prow workload clusters and specify the image URLs for the prowjobs ([link](https://github.com/kyma-project/test-infra/pull/7882/files?diff=unified&w=0#diff-7746f51f47dc4a9ed5457c61d4dd1ceb829d3ea736f8ad83e8c5efb4432246f6R1-R33))
*  Apply the `k8sallowedimages` template to two other namespaces in the Prow workload clusters and specify the image URLs for the supporting tools ([link](https://github.com/kyma-project/test-infra/pull/7882/files?diff=unified&w=0#diff-932f92c0aedb6b13d14e5d3f324f4b17bc1f4dc8a07fd7f98df28b64a3d48952R1-R23))

### CODE REVIEW

File: prow/cluster/resources/gatekeeper-constraints/workloads/allowedImagesProwToolsNamespaces.yaml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+metadata:
+  name: prow-tools-namespaces
+spec:
+  enforcementAction: warn
+  match:
+    kinds:
+      - apiGroups: [""]
+        kinds: ["Pod"]
+  parameters:
+    namespaces:
```
Comment (halamix2) [inline prow/cluster/resources/gatekeeper-constraints/workloads/allowedImagesProwToolsNamespaces.yaml:14]: namespaces can be defined in the `spec.match`, they don't need to even be in the constraint template

**Review by halamix2 — CHANGES_REQUESTED** [review]

File: prow/cluster/resources/gatekeeper-constraints/workloads/allowedImagesProwToolsNamespaces.yaml (hunk 1 above)
Comment (halamix2) [reply prow/cluster/resources/gatekeeper-constraints/workloads/allowedImagesProwToolsNamespaces.yaml:14]: Example https://github.com/kyma-project/test-infra/blob/main/prow/cluster/resources/gatekeeper-constraints/workloads/apparmor.yaml#L11

Comment (dekiel) [reply prow/cluster/resources/gatekeeper-constraints/workloads/allowedImagesProwToolsNamespaces.yaml:13]: corrected

**Review by halamix2 — APPROVED** [review]

**Pull request merged**